### PR TITLE
Remaining functionality for new library and loadout pages

### PR DESCRIPTION
**Pull request by erri120**
- adds the remove button to the new library page
- adds a filter to the new library and loadout pages
- fixed an issue with row selection

TODO:
- [x] filter by game in the library
- [x] created at for mod pages should show file created at

### CODE REVIEW

**Review by Al12rs — COMMENTED** [review]
The parent here in the Loadout view should be using the most recent `Installed` value from the children:
image

**Review by Al12rs — COMMENTED** [review]
This is also very strange:

Added column says 3days ago, I did have this installed and I removed it, but it should not be picking that up right?
image

File: src/NexusMods.App.UI/Pages/NexusModsDataProvider.cs
Comment (Al12rs) [inline src/NexusMods.App.UI/Pages/NexusModsDataProvider.cs:178]: This causes exception on deletes, as it can become an empty collection. Should check if Items is empty and return the default in that case.

File: src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridItemModel.cs
Comment (Al12rs) [inline src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridItemModel.cs:120]: This isn't working in case one of the children is removed while the parent is collapsed.
We still have the child, but when we start processing updates again we only receive Adds for every element in the current collection, no Removes for our existing items since we had stopped listening.

So we would need to have special handling for when we make the subscription that checks the current SourceCache against the current contents and updates, then goes to listen changes from there.

**Review by Al12rs — APPROVED** [review]
Code looks good, need to wait for QA approval.

**Review by Al12rs — CHANGES_REQUESTED** [review]
Somehow managed to break selection again. Will need a bit more testing.

**Review by Al12rs — APPROVED** [review]
I couldn't break it anymore

**Review by Al12rs — CHANGES_REQUESTED** [review]
Never mind, deleting from Library stops working after a bit

**Comment by erri120** [comment]
> Never mind, deleting from Library stops working after a bit

I'd say that's unrelated.

**Review by Al12rs — APPROVED** [review]
Yeah, haven't been able to reproduce, not for lack of trying.
We will keep an eye out to see if it happens again in the future.

**Pull request merged**
